README.md

```diff
@@ -108,6 +108,45 @@ python -m SimpleHTTPServer
 curl -v -p --proxy-key certs/master/private/proxy-client.key --proxy-cert certs/master/issued/proxy-client.crt --proxy-cacert certs/master/issued/ca.crt --proxy-cert-type PEM -x https://127.0.0.1:8090  http://localhost:8000```
 ```
 
+### GRPC Client using mTLS Proxy with dial back Agent runs on k8s cluster
+
+- Push all images to container registry
+```console
+TAG=1.1
+make docker-push TAG=${TAG}
+```
+
+- Start test web-server as kubia pod and service
+```console
+kubectl apply -f artifacts/k8s-templates/kubia.yaml
+```
+
+- Initialize environment variables (replace content of CLUSTER_CERT and CLUSTER_KEY that matches kubeapi-server's certificates location)
+```console
+CLUSTER_IP=$(kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}' | sed -n "s/https\:\/\/\(\S*\).*$/\1/p")
+KUBIA_IP=$(kubectl get svc kubia -o=jsonpath='{.spec.clusterIP}')
+PROXY_IMAGE=$(docker images | grep "proxy-server-" -m1 | awk '{print $1}')
+AGENT_IMAGE=$(docker images | grep "proxy-agent-" -m1 | awk '{print $1}')
+TEST_CLIENT_IMAGE=$(docker images | grep "proxy-test-client-" -m1 | awk '{print $1}')
+CLUSTER_CERT=/etc/srv/kubernetes/pki/apiserver.crt
+CLUSTER_KEY=/etc/srv/kubernetes/pki/apiserver.key
+```
+
+- Start **proxy-server** as a [static pod](https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/) with following configuration
+```console
+TAG=${TAG} PROXY_IMAGE=${PROXY_IMAGE} CLUSTER_CERT=${CLUSTER_CERT} CLUSTER_KEY=${CLUSTER_KEY} envsubst <  artifacts/k8s-templates/konnectivity-proxy.yaml
+```
+
+- Start **proxy-agent** as a k8s pod
+```console
+TAG=${TAG} AGENT_IMAGE=${AGENT_IMAGE} CLUSTER_IP=${CLUSTER_IP}  envsubst <  artifacts/k8s-templates/konnectivity-agent.yaml | kubectl apply -f -
+```
+
+- Run **test-client** as following command on master VM. (last row needs to be **You've hit kubia**)
+```console
+echo "sudo proxy-test-client --proxy-uds=/etc/srv/kubernetes/konnectivity/konnectivity-server.socket   --proxy-host='' --proxy-port=0 --request-port=80 --request-host=${KUBIA_IP}"
```

Comment:
Can we just use a curl client instead of `proxy-test-client`? It feels like we're adding a lot of additional stuff to the build

Reply:
The whole idea of this client to test all possible permutation of server. With self signed certificates / kubeapi-server certificates / uds. If we replace the client with curl, we need introduce a new kind of  listener on proxy that will be used for tests only

Reply:
Discussed offline, we should test http-connect instead of grpc for now. Should be fine to keep the `proxy-test-client` since it simplifies a lot of the testing.

Reply:
I tested code by adding "--mode=http-connect" to the konnectivity-proxy.yaml and adding "--mode=http-connect" to proxy-test-client.
I still not able to configure curl proxy to work with UDS socket

Reply:
sudo curl -v -p     --unix-socket /etc/srv/kubernetes/konnectivity/konnectivity-server.socket    -x http://10.75.246.169:80
curl: no URL specified!
curl: try 'curl --help' or 'curl --manual' for more information

Reply:
Typo in the heading. This is UDS + http-connect. I've tested this on my end as well.

Reply:
I modified the  example to be UDS + http-connect